### PR TITLE
Support for Raw data to MDM Transformer class with Implementation for MISO

### PR DESCRIPTION
**Pull request by IW-SS**
The PR contains the following changes - 

1. Query timezone support for MISO Source + Test cases
2. Generic transformer class (BaseRawToMDMTransformer) to convert a raw data into MDM (Meters Data Model).
3. Implementation of BaseRawToMDMTransformer class for MISO source + Test cases + Documentation.

### CODE REVIEW

File: src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from pyspark.sql.types import StructType, StructField, TimestampType, DoubleType, StringType, MapType, IntegerType
+
+MDM_USAGE_SCHEMA = StructType([
+    StructField("uid", StringType(), True),
+    StructField("series_id", StringType(), True),
+    StructField("timestamp", TimestampType(), True),
+    StructField("interval_timestamp", TimestampType(), True),
+    StructField("value", DoubleType(), True)
+])
+
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py:24]: To be consistent across Models, can we change these column names to Cased Names? Such as SeriesId, TimestampStart, ModelType etc?

File: src/sdk/python/rtdip_sdk/pipelines/sources/spark/iso/miso_daily_load_iso.py [hunk 2]
```diff
@@ -48,6 +47,7 @@ class MISODailyLoadISOSource(BaseISOSource):
     query_datetime_format: str = "%Y%m%d"
     required_options = ["load_type", "date"]
     spark_schema = MISO_SCHEMA
+    default_query_timezone = "US/Central"
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/sources/spark/iso/miso_daily_load_iso.py:50]: Should this be UTC instead?

File: src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+MDM_META_SCHEMA = StructType([
+    StructField("uid", StringType(), True),
+    StructField("series_id", StringType(), True),
+    StructField("series_parent_id", StringType(), True),
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py:28]: As above, can these be renamed to comply with cased names?

File: src/sdk/python/rtdip_sdk/pipelines/sources/spark/iso/miso_daily_load_iso.py [hunk 4]
```diff
@@ -131,10 +131,13 @@ def _validate_options(self) -> bool:
         """
 
         try:
-            datetime.strptime(self.date, self.query_datetime_format)
+            date = self._get_localized_datetime(self.date)
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/sources/spark/iso/miso_daily_load_iso.py:134]: Should this be UTC?

File: src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py (hunk 1 above)
Comment (IW-SS) [reply src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py:24]: Changed to CamelCasing.

File: src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py (hunk 3 above)
Comment (IW-SS) [reply src/sdk/python/rtdip_sdk/pipelines/_pipeline_utils/mdm.py]: Changed to CamelCasing.

**Review by GBBBAS — APPROVED** [review]

**Review by cching95 — APPROVED** [review]

**Pull request merged**